src/test/java/streams/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,139 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import lombok.Value;
+import org.junit.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise1 {
+
+    @Test
+    public void findPersonsEverWorkedInEpam() {
+        List<Employee> employees = getEmployees();
+
+        List<Person> personsEverWorkedInEpam = employees
+                .stream()
+                .flatMap(employee
+                        -> employee
+                        .getJobHistory()
+                        .stream()
+                        .map(jobHistoryEntry -> new PersonJobHistoryEntryPair(employee.getPerson(), jobHistoryEntry)))
+                .filter(personJobHistoryEntryPair -> "EPAM".equals(personJobHistoryEntryPair.getJobHistoryEntry().getEmployer()))
+                .map(PersonJobHistoryEntryPair::getPerson)
+                .distinct()
+                .collect(Collectors.toList());
```

Comment:
You can do this simplier:
1. ```filter``` employes that at ```getJobHistroy().stream()``` have any match with employee "EPAM"
2. map them into persons
3. collect to list